src/routing/approved.rs

```diff
@@ -1005,8 +1002,16 @@ impl Approved {
                 message: MessageType::ClientMessage(ClientMessage::from(content)?),
             }]);
         }
-        if let DstLocation::AccumulatingNode(_name) = &dst {
-            if let SrcAuthority::BlsShare { proof_share, .. } = &src {
+        if msg.aggregate_at_dst() {
+            if !matches!(dst, DstLocation::Node(_)) {
+                return Err(Error::InvalidDstLocation);
+            }
```

Comment:
Why is this an error? Can't we have a dst-accumulated message sent to a section?

Reply:
Mm, mistake, should include Section as well.